hazelcast/src/main/java/com/hazelcast/sql/SqlService.java

```diff
@@ -16,81 +16,106 @@
-
 import javax.annotation.Nonnull;
 import java.util.List;
 
 /**
  * A service to execute SQL statements.
  * <p>
  * In order to use the service, Jet engine must be enabled - SQL statements are executed as
- * Jet jobs. On members, the {@code hazelcast-sql.jar} must be on the member's classpaths,
- * otherwise an exception will be thrown; on client, it is not necessary.
+ * Jet jobs. On members, the {@code hazelcast-sql.jar} must be on the classpath, otherwise
+ * an exception will be thrown; on client, it is not necessary.
  * <p>
  * The text below summarizes features supported by the SQL engine.
- *
+ * <p>
  * <h1>Overview</h1>
  * Hazelcast is able to execute distributed SQL queries over the following entities:
  * <ul>
  *     <li>IMap
+ *     <li>Kafka
+ *     <li>Files
  * </ul>
- * When an SQL statement is submitted to a member, it is parsed and optimized by the {@code hazelcast-sql} module, that is based
- * on <a href="https://calcite.apache.org">Apache Calcite</a>. During optimization a statement is converted
- * into a directed acyclic graph (DAG) that is sent to cluster members for execution.
- * Results are sent back to the originating member asynchronously and returned to the user via {@link SqlResult}.
- *
+ * When an SQL statement is submitted to a member, it is parsed and optimized by the {@code hazelcast-sql} module,
+ * that is based on <a href="https://calcite.apache.org">Apache Calcite</a>. During optimization a statement is
+ * converted into a directed acyclic graph (DAG) that is sent to cluster members for execution. Results are sent
+ * back to the originating member asynchronously and returned to the user via {@link SqlResult}.
+ * <p>
  * <h1>Querying an IMap</h1>
- * Every IMap instance is exposed as a table with the same name in the {@code partitioned} schema. The {@code partitioned}
- * schema is included into a default search path, therefore an IMap could be referenced in an SQL statement with or without the
- * schema name.
- * <h2>Column resolution</h2>
- * Before you can access an IMap, a <em>mapping</em> has to be created first. See the reference manual
- * for the CREATE MAPPING command.
+ * Before you can access an IMap, a <em>mapping</em> has to be created first. See the reference manual for the
+ * CREATE MAPPING command.
  * <p>
- * Columns are extracted from objects as follows:
+ * <h2>Serialization Options</h2>
+ * The <em>keyFormat</em> and <em>valueFormat</em> options are mandatory. Possible values for <em>keyFormat</em>
+ * and <em>valueFormat</em>:
  * <ul>
- *     <li>For non-Portable objects, public getters and fields are used to populate the column list. For getters, the first
- *     letter is converted to lower case. A getter takes precedence over a field in case of naming conflict
- *     <li>For {@link Portable} objects, field names used in the {@link Portable#writePortable(PortableWriter)} method
- *     are used to populate the column list
+ *     <li>java
```

Comment:
Maybe worth mentioning here that you can also specify any primitive types as well? E.g. `bigint`/`varchar` etc.

Reply:
I'd remove this and the following sections altogether - i don't like documentation duplication. It's described in the refperson. On the other hand I'd mention here that queries are not atomic (can fail and commit part of the data).